packages/neo-one-client-core/src/user/keystore/HDKeyStore.ts

```diff
@@ -0,0 +1,224 @@
+import {
+  Account,
+  AddressString,
+  NetworkType,
+  PublicKeyString,
+  UserAccount,
+  UserAccountID,
+} from '@neo-one/client-common';
+import { Monitor } from '@neo-one/monitor';
+import { mergeScanLatest, utils } from '@neo-one/utils';
+import _ from 'lodash';
+import { BehaviorSubject, Observable, Subscription } from 'rxjs';
+import { distinctUntilChanged, map } from 'rxjs/operators';
+import { UnknownAccountError, UnknownNetworkError } from '../../errors';
+import { KeyStore } from '../LocalUserAccountProvider';
+
+export interface HDAccount<Identifier> {
+  readonly identifier: Identifier;
+  readonly userAccount: UserAccount;
+}
+
+type HDAccounts<Identifier> = {
+  readonly [Network in string]?: { readonly [Address in string]?: HDAccount<Identifier> }
+};
+
+export interface HDProvider {
+  readonly getAccount: (network: NetworkType, address: AddressString, monitor?: Monitor) => Promise<Account>;
+  readonly getNetworks: () => ReadonlyArray<NetworkType>;
+  readonly networks$: Observable<ReadonlyArray<NetworkType>>;
+}
+
+export interface HDStore<Identifier> {
```

Comment:
Is this interface necessary? Seems like we should be using `HDHandler` everywhere instead?

Reply:
Hmm, maybe this is just a common type you're using? Move it into a `types.ts` file since it's not used anywhere in this file.